stablehlo/tests/interpret_bitcast_convert.mlir

```diff
@@ -0,0 +1,278 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
+
+func.func @bitcast_convert_op_test_i1_to_i1() {
```

Comment:
The tests covers the following non complex types like `i1`, `i64`, `i4`, `ui64`, `f64`, `f16` `bf16`.

Given that the new proposal https://github.com/ghpvnist/stablehlo/pull/8 decouples `bitcastConvertManyToOne`, `bitcastConvertOneToMany` and `bitcastConvertOneToOne`  from  handling individual type conversions and rather puts all the type handling logic in   `fromBits` and `toBits`, do we need to use all the tests here? 

From the code coverage point of view, may be we need to cover  function like `fromBits` and `toBits` while writing tests with different types. For example, we may use tests to cover non-complex types

```
// to cover the non-complex types in fromBits

bitcast_convert_op_test_i1_to_i64
bitcast_convert_op_test_i64_to_i64
bitcast_convert_op_test_f64_to_i64

// to cover the non-complex types in toBits

bitcast_convert_op_test_i64_to_i1
bitcast_convert_op_test_i64_to_i64
bitcast_convert_op_test_i64_to_f64
```

what do you think?

Reply:
We need to cover test conversions for:
* `any type` to `any type` (not necessarily every combination).
* many to one
* one to many
* one to one

The following three tests has full coverage (I chose `bf16` for shorter tests compared to `i1`):
bf16 -> i64
i64 -> f64
f64 -> bf16

Reply:
The suggestion also looks good except that the we need a test for `i1` as source and destination. This is because boolean type is handled explicitly in both `from_bits` and `to_bits`. 

What about we modify the suggestion with replacing `bf16` with `i1`. Which that we still cover all of many to one,
one to many, and one to one. Plus test `from_bits` (and `to_bits`) with `bool`, `int`, `float`, `complex`  as source (and destination)

```
i1 -> i64
i64 -> f64
f64 -> i1
c128 -> c64
```

Reply:
Sure, makes sense.